lightbluetent/rooms.py

```diff
@@ -174,32 +177,116 @@ def update(room_id, update_type):
             room.name = values["name"]
             room.authentication = Authentication(values["authentication"])
             room.description = values["description"]
-            print(room.description)
             room.alias = values["alias"] if values["alias"] != "" else None
 
 
     elif update_type == "room_times":
-        keys = (
-            "start_date",
-            "start_hour",
-            "start_min",
-            "end_date",
-            "end_hour",
-            "end_min"
-        )
 
         # Is there a better way to do this?
-        start_date_entered = values["start_date"] != "" or values["start_hour"] != "" or values["start_min"] != ""
-        end_date_entered = values["end_date"] != "" or values["end_hour"] != "" or values["end_min"] != ""
-        full_start_date = values["start_date"] != "" and values["start_hour"] != "" and values["start_min"] != ""
-        full_end_date = values["end_date"] != "" and values["end_hour"] != "" and values["end_min"] != ""
+        valid_start_date = validate_date(values["start_date"]) and values["start_hour"].isdigit() and values["start_min"].isdigit()
+        valid_end_date = validate_date(values["end_date"]) and values["end_hour"].isdigit() and values["end_min"].isdigit()
+
+        if not valid_start_date:
+            errors["start"] = "You must specify a starting time and date."
+        elif not valid_end_date:
+            errors["end"] = "You must specify a ending time and date."
+        else:
+            if int(values["start_hour"]) > 24 or int(values["start_min"]) > 60:
+                errors["start"] = "Invalid start time."
+            if int(values["end_hour"]) > 24 or int(values["end_min"]) > 60:
+                errors["end"] = "Invalid end time."
+
+            start_date = values["start_date"].split("-")
+            start_year = int(start_date[0])
+            start_month = int(start_date[1])
+            start_day = int(start_date[2])
+
+            end_date = values["end_date"].split("-")
+            end_year = int(end_date[0])
+            end_month = int(end_date[1])
+            end_day = int(end_date[2])
+
+            start = datetime(start_year, start_month, start_day, int(values["start_hour"]), int(values["start_min"]))
```

Comment:
Does datetime not accept YYYY-MM-DD format? It seems it would be so common that it would just parse it itself, same with the time.